### PR TITLE
move getTransformTo method from RenderBox to RenderObject.

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

**Comment by jacob314** [comment]
No changes to the method other than promoting it to the base class as it is convenient for all RenderObject instances.

File: packages/flutter/lib/src/rendering/object.dart [hunk 1]
```diff
@@ -2341,6 +2341,35 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     assert(child.parent == this);
   }
 
+  /// Returns a matrix that maps the local coordinate system to the coordinate
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: As documented, this doesn't really make sense. e.g. RenderSlivers and RenderSectors don't use the cartesian coordinate system for layout, so it's not clear what a Matrix4 means.

The fix, I think, is just to change the documentation to clarify that this refers to the _paint_ coordinate system, and that specifically this is just applying `applyPaintTransform` up the tree.

**Comment by Hixie** [comment]
Can you add a test that this works for non-box render objects?

With a test and the fix to the documentation, LGTM.

File: packages/flutter/lib/src/rendering/object.dart (hunk 1 above)
Comment (jacob314) [inline packages/flutter/lib/src/rendering/object.dart]: Updated comment.

**Comment by jacob314** [comment]
Added a test and updated the comment.

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
@jacob314 Is this ready to land?

**Comment by Hixie** [comment]
There seems to be no reason not to land this so I'm landing it.

**Pull request merged**